examples/step-82/step-82.cc

```diff
@@ -0,0 +1,1500 @@
+                               local_dof_indices[j]) += ip_matrix_nc(i, j);
+                        matrix(local_dof_indices_neighbor[i],
+                               local_dof_indices_neighbor[j]) +=
+                          ip_matrix_nn(i, j);
+                      }
+                  }
+              }
+
+          } // for face
+      }     // for cell
+  }
+
+
+
+  // @sect4{BiLaplacianLDGLift::assemble_rhs}
+
+  // This function assemble the right-hand side of the system. Since we consider
+  // homogeneous Dirichlet boundary conditions, imposed weakly in the bilinear
+  // form using the Nitsche approach, it only involves the contribution of the
+  // forcing term $\int_{\Omega}fv_h$.
+  template <int dim>
+  void BiLaplacianLDGLift<dim>::assemble_rhs()
+  {
+    rhs = 0;
+
+    const QGauss<dim> quad(fe.degree + 1);
+    FEValues<dim>     fe_values(
+      fe, quad, update_values | update_quadrature_points | update_JxW_values);
+
+    const unsigned int n_dofs     = fe_values.dofs_per_cell;
+    const unsigned int n_quad_pts = quad.size();
+
+    const RightHandSide<dim> right_hand_side;
+
+    Vector<double>                       local_rhs(n_dofs);
+    std::vector<types::global_dof_index> local_dof_indices(n_dofs);
+
+    for (const auto &cell : dof_handler.active_cell_iterators())
+      {
+        fe_values.reinit(cell);
+        cell->get_dof_indices(local_dof_indices);
+
+        local_rhs = 0;
+        for (unsigned int q = 0; q < n_quad_pts; ++q)
+          {
+            const double dx = fe_values.JxW(q);
+
+            for (unsigned int i = 0; i < n_dofs; ++i)
+              {
+                local_rhs(i) +=
+                  right_hand_side.value(fe_values.quadrature_point(q)) *
+                  fe_values.shape_value(i, q) * dx;
+              }
+          }
+
+        for (unsigned int i = 0; i < n_dofs; ++i)
+          rhs(local_dof_indices[i]) += local_rhs(i);
+      }
+  }
+
+
+
+  // @sect4{BiLaplacianLDGLift::solve}
+
+  // To solve the linear system $A\boldsymbol{U}=\boldsymbol{F}$,
+  // we proceed as in step-74 and use a direct method. We could
+  // as well use an iterative method, for instance the conjugate
+  // gradient method as in step-3.
+  template <int dim>
+  void BiLaplacianLDGLift<dim>::solve()
+  {
+    SparseDirectUMFPACK A_direct;
+    A_direct.initialize(matrix);
+    A_direct.vmult(solution, rhs);
+  }
+
+
+
+  // @sect4{BiLaplacianLDGLift::compute_errors}
+
+  // This function computes the discrete $H^2$, $H^1$ and $L^2$ norms of
+  // the error $u-u_h$, where $u$ is the exact solution and $u_h$ is
+  // the approximate solution. See the introduction for the definition
+  // of the norms.
+  template <int dim>
+  void BiLaplacianLDGLift<dim>::compute_errors()
+  {
+    double error_H2 = 0;
+    double error_H1 = 0;
+    double error_L2 = 0;
+
+    QGauss<dim>     quad(fe.degree + 1);
+    QGauss<dim - 1> quad_face(fe.degree + 1);
+
+    FEValues<dim> fe_values(fe,
```

Comment:
I guess ones cannot use the `VectorTools` functions here because they do not consider faces(?). Would it be an option to create a new function in that namespace?